internal/api/utils.go

```diff
@@ -370,5 +371,64 @@ func TransformComponentUnits(jsonData datatypes.JSON) map[string]interface{} {
 		}
 	}
 
+	return recommendationJSON
+}
+
+func filterNotifications(recommendationID string, clusterUUID string, recommendationJSON map[string]interface{}) map[string]interface{} {
+
+	deleteNotificationObject := func(recommendationSection map[string]interface{}) {
+		notificationObject, ok := recommendationSection["notifications"].(map[string]interface{})
+		if ok {
+			for key := range notificationObject {
+				_, found := NotificationsToShow[key]
+				if !found {
+					delete(recommendationSection, "notifications")
+					log.Warnf("dropped notification %s dropped from recommendation ID: %s; cluster ID: %s", key, recommendationID, clusterUUID)
+				}
+			}
+		}
+
+	}
+
+	// level 1 notifications are not stored in the database
+
+	// level 2
+	deleteNotificationObject(recommendationJSON)
+
+	recommendationTerms, ok := recommendationJSON["recommendation_terms"].(map[string]interface{})
+	if !ok {
+		log.Error("recommendation data not found in JSON")
+		return recommendationJSON
+	}
+
+	for _, term := range []string{"short_term", "medium_term", "long_term"} {
+		levelThree, ok := recommendationTerms[term].(map[string]interface{})
+		if ok {
+			deleteNotificationObject(levelThree)
+		}
+		recommendationEngineObject, ok := levelThree["recommendation_engines"].(map[string]interface{})
```

Comment:
If I see the [kruize docs](https://github.com/kruize/autotune/blob/remote_monitoring/design/NotificationCodes.md) then all of the recommendation codes are of engine level, so should we not just remove that from engine level?

Reply:
I think you are talking about `NotificationsToShow` codes